src/App.js

```diff
@@ -1,75 +1,131 @@
 import React from 'react';
+import NewTodo from './NewTodo';
+import TodoList from './TodoList';
+import Footer from './Footer';
 
-function App() {
-  return (
-    <section className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
+class App extends React.Component {
+  state = {
+    todos: [],
+    filters: ['all', 'active', 'completed'],
+    valueTitle: '',
+    filterName: '',
+    activeFilterIndex: 0,
+  }
 
-        <input
-          className="new-todo"
-          placeholder="What needs to be done?"
-        />
-      </header>
+  selectAllHandler = () => {
+    this.setState((state) => {
+      const allSelected = state.todos.every(todo => todo.completed);
 
-      <section className="main" style={{ display: 'block' }}>
-        <input type="checkbox" id="toggle-all" className="toggle-all" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
+      return {
+        todos: state.todos.map(todo => ({
+          ...todo,
+          completed: !allSelected,
+        })),
+      };
+    });
+  }
 
-        <ul className="todo-list">
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-1" />
-              <label htmlFor="todo-1">sdfsdfsdf</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
+  InputChangeHandler = (event) => {
+    this.setState({
+      valueTitle: event.target.value,
+    });
+  };
 
-          <li className="">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="todo-2" />
-              <label htmlFor="todo-2">sakgjdfgkhjasgdhjfhs</label>
-              <button type="button" className="destroy" />
-            </div>
-          </li>
+  addTodoHandler = (event) => {
```

Comment:
a чому addTodo обробляє івенти? З назви метода вона має приймати todo і додавати його в масив todo
```suggestion
  addTodo = (todo) => this.setState(prev => ({
    todos: [...prev.todos, todo],
    selectedFilter: FILTER_TYPES.all,
  }))
```
Ось так вона має виглядати і нічого більше:)

Reply:
Тоесть на обработчик onSubmit мне придется ставить 2 отдельные функции : одну, которая создает todo на основании введенного в инпуте формы title и другую, которая добавляет эту todo в массив. Зачем тогда в этом случае делать 2 функции, если функция создания todo с параметром title неразрывно связана с его добавлением в массив и дальнейши отображением?